src/components/Pagination/index.js

```diff
@@ -0,0 +1,366 @@
+import styled from '@emotion/styled'
+import PropTypes from 'prop-types'
+import React, {
+  createContext,
+  forwardRef,
+  useCallback,
+  useContext,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import ActivityIndicator from '../ActivityIndicator'
+import Box from '../Box'
+import Button from '../Button'
+import getPageNumbers from './getPageNumbers'
+import usePaginatedData from './usePaginatedData'
+
+const PaginationContext = createContext()
+export const usePagination = () => useContext(PaginationContext)
+
+// START - Default Components
+const DefaultLeftComponent = () => null
+
+const StyledPageButton = styled(Button, {
+  shouldForwardProp: prop => !['current'].includes(prop),
+})`
+  ${({ current }) =>
+    current
+      ? `
+    text-decoration: underline;
+    &:hover {
+      text-decoration: underline;
+    }
+  `
+      : ''}
+`
+
+const DefaultMiddleComponent = ({ pageTabCount }) => {
+  const {
+    isLoadingPage,
+    page,
+    maxPage,
+    canLoadMore,
+    goToNextPage,
+    goToFirstPage,
+    goToLastPage,
+    goToPage,
+    goToPreviousPage,
+  } = usePagination()
+
+  const pageNumbersToDisplay = useMemo(
+    () => getPageNumbers(page, maxPage, pageTabCount),
+    [page, maxPage, pageTabCount],
+  )
+
+  const handlePageClick = useCallback(
+    pageNumber => () => {
+      goToPage(pageNumber)
+    },
+    [goToPage],
+  )
+
+  return (
+    <div>
+      <Button
+        mr={1}
+        disabled={page === 1 || isLoadingPage}
+        onClick={goToFirstPage}
+        aria-label="First"
+      >
+        First
+      </Button>
+      <Button
+        aria-label="Back"
+        mr={1}
+        disabled={page === 1 || isLoadingPage}
+        onClick={goToPreviousPage}
+      >
+        Back
+      </Button>
+      {pageNumbersToDisplay.map(pageNumber => (
+        <StyledPageButton
+          aria-label={`Page ${pageNumber}`}
+          key={pageNumber}
```

Comment:
Maybe add a prefix here, if there is another component with same key it will  create problems